cobc/typeck.c

```diff
@@ -221,7 +221,7 @@ static const char	*const bin_sub_funcs[] = {
 	"cob_subswp_s64_binary"
 };
 
-static const char	*const align_bin_compare_funcs[] = {
+static const char	*const align_bin_compare_funcs[] __attribute((unused)) = {
```

Comment:
Instead of adding \_\_attribute\_\_((unused)), it is better to remove align_bin_compare_funcs.

\_\_attribute\_\_((unused)) is a non-standard feature that is available in gcc, but not necessarily in other C compilers. For example, it does not seem to be available in Visual Studio. https://stackoverflow.com/questions/52058457/visual-studio-equivelent-of-gcc-attribute-unused-in-c11-or-lower

Although rarely used, opensource COBOL may be used on Windows. For example, libcob/fileio.c describes the process for Windows.
https://github.com/yutaro-sakamoto/opensource-cobol/blob/93def13076e056d9a31be701b0ae080fce02f6d7/libcob/fileio.c#L59-L67

The use of \_\_attirbute\_\_((unused)) may reduce portability.
It is a good idea to remove unused variables and functions as it should not affect the operation.

Remove unused variables and see if the NIST test succeeds.

The same applies to other code where \_\_attirbute\_\_((unused)) is used.